api/v1beta1/policyautomation_types.go

```diff
@@ -94,3 +114,22 @@ type ClusterEvent struct {
 func init() {
 	SchemeBuilder.Register(&PolicyAutomation{}, &PolicyAutomationList{})
 }
+
+// ReplicatedDetailsPerTemplate defines the replicated policy compliance details and history
+type ReplicatedDetailsPerTemplate struct {
+	ComplianceState policyv1.ComplianceState      `json:"compliant"`
+	History         []ReplicatedComplianceHistory `json:"history"`
+}
+
+// ReplicatedComplianceHistory defines the replicated policy compliance details history
+type ReplicatedComplianceHistory struct {
+	LastTimestamp metav1.Time `json:"lastTimestamp,omitempty" protobuf:"bytes,7,opt,name=lastTimestamp"`
+	Message       string      `json:"message,omitempty" protobuf:"bytes,4,opt,name=message"`
+}
+
+// ReplicatedPolicyStatus defines the replicated policy status
+type ReplicatedPolicyStatus struct {
+	ComplianceState  policyv1.ComplianceState        `json:"compliant"`        // used by replicated policy
+	ViolationMessage string                          `json:"violationMessage"` // used by replicated policy
+	Details          []*ReplicatedDetailsPerTemplate `json:"details"`          // used by replicated policy
```

Comment:
We want an empty list in this case when converted to an extra var. That way I don't have check the type in my Ansible playbook. I can just loop over it.
```suggestion
	Details          []ReplicatedDetailsPerTemplate `json:"details"`          // used by replicated policy
```